### PR TITLE
Add documentation for Scala 3 derivation

### PR DESCRIPTION
**Pull request by ruippeixotog**
As titled.

@heksesang, if you could take a quick look it would be great.

### CODE REVIEW

File: docs/docs/docs/scala-3-derivation.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+Scala 3 introduced native support for
+[type class derivation](https://dotty.epfl.ch/docs/reference/contextual/derivation.html) - a way to automatically
+generate type class instances for enums, enum cases, case classes, case objects and sealed traits that only have the
+aforementioned types as children. Instances are generated on demand by adding a `derives` clause on data types.
+
+PureConfig has some limited support for derivation of `ConfigReader` instances. In order to enable it, the relevant
+extensions must be imported:
+
+```scala
```
Comment (ruippeixotog) [inline docs/docs/docs/scala-3-derivation.md:16]: Unfortunately we can't run any of this through mdoc as our website depends on `pureconfig-generic`, which isn't available in Scala 3. Instead, I ran this on `core/console` and copied the outputs here.

File: docs/docs/docs/scala-3-derivation.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import pureconfig._
+import pureconfig.generic.derivation.default._
+```
+
+After that, you can derive `ConfigReader` instaces for your config class using a `derives` clause:
```
Comment (heksesang) [inline docs/docs/docs/scala-3-derivation.md]: Missing an n in instances here.

**Comment by heksesang** [comment]
LGTM, except that one typo I pointed out. :)

File: docs/docs/docs/scala-3-derivation.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+### Limitations
+
+There is currently no way to customize case class and sealed trait derivation - any `ProductHint` and `CoproductHint`
+instances in scope are ignored. 
+
+`ConfigWriter` derivation is not supported at the moment.
+
+Overall, consider this to be in alpha/beta stage. This implementation is not as mature as the Shapeless or
+Magnolia-based derivation libaries available in Scala 2 yet so please report any bugs that you find!
```
Comment (jcazevedo) [inline docs/docs/docs/scala-3-derivation.md]:
```suggestion
Magnolia-based derivation libraries available in Scala 2 yet so please report any bugs that you find!
```

**Review by jcazevedo — CHANGES_REQUESTED** [review]
Other than a small typo, this looks good to me!

**Review by jcazevedo — APPROVED** [review]
Thanks!

**Pull request merged**